homeassistant/components/sensor/abode.py

```diff
@@ -0,0 +1,77 @@
+"""
+This component provides HA sensor support for Abode Security System.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.abode/
+"""
+import logging
+
+from homeassistant.components.abode import AbodeDevice, DOMAIN as ABODE_DOMAIN
+
+
+DEPENDENCIES = ['abode']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Sensor types: Name, icon
+SENSOR_TYPES = {
+    'temp': ['Temperature', 'thermometer'],
+    'humidity': ['Humidity', 'water-percent'],
+    'lux': ['Lux', 'lightbulb'],
+}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up a sensor for an Abode device."""
+    import abodepy.helpers.constants as CONST
+
+    data = hass.data[ABODE_DOMAIN]
+
+    devices = []
+    for device in data.abode.get_devices(generic_type=CONST.TYPE_SENSOR):
+        if data.is_excluded(device):
+            continue
+
+        for sensor_type in SENSOR_TYPES:
+            devices.append(AbodeSensor(data, device, sensor_type))
+
+    data.devices.extend(devices)
+
+    add_devices(devices)
+
+
+class AbodeSensor(AbodeDevice):
+    """A sensor implementation for Abode devices."""
+
+    def __init__(self, data, device, sensor_type):
+        """Initialize a sensor for an Abode device."""
+        super().__init__(data, device)
+        self._sensor_type = sensor_type
+        self._icon = 'mdi:{}'.format(SENSOR_TYPES[self._sensor_type][1])
+        self._name = "{0} {1}".format(self.name,
```

Comment:
This isn't used, it seems?

Reply:
Forgive me, Python isn't my strong suite. If my super class sets self._name with a property to return it, and the subclass overrides that same value, would the superclass property return the subclasses overridden value?

Reply:
Yes, but the superclass property does not return `self._name` in this case. It returns `self.device.name`.

Reply:
D'oh!